R/klass.R

```diff
@@ -0,0 +1,146 @@
+#' Get municipalities
+#'
+#' Get all Norwegian municipalities and municipality codes for a given year.
+#'
+#' Fetches all Norwegian municipalities and municipality codes for a given year.
+#' All years from 1977 until present are supported. The function will default to
+#' the current year. The function will return a data.frame with all municipalities.
+#' If more than one year is given, a single data.frame will be returned with all
+#' codes and names for the given years, unless `simplify` is set to `FALSE`, then
+#' a list of the same length of the input will be returned instead.
+#'
+#' You can add county numbers and names for each municipality by setting `add_county`
+#' to `TRUE`.
+#'
+#' @inheritParams get_entity
+#' @param year The year for which the codes should be valid for
+#' @param add_county If set to `TRUE`, add county name and number
+#' @param include_notes If notes should be included or not
+#' @param simplify If `TRUE` (default), a single data.frame is returned.
+#'
+#' @return data.frame or list
+#'
+#' @seealso `get_counties()` for county codes and names
+#'
+#' @export
+get_municipalities <- function(
+    year = format(Sys.Date(), '%Y'), add_county = FALSE, include_notes = FALSE,
+    simplify = TRUE, raw_response = FALSE)
+{
+  if (raw_response) simplify <- FALSE
```

Comment:
Actually, based on the logic we have employed in these functions, `simplify` and `raw_response` will be mutually exclusive. Setting `simplify = TRUE` will return a single `data.frame` from the function using `rbind()`. However, if you set `raw_response = TRUE`, the function will always return a list object, as the raw response is set in `get_klass_codes_single()`. It will not be possible to collapse the returned data.frames when `raw_response = TRUE` without a major rewrite of the function itself.

Reply:
@ohjakobsen: As discussed I think we can live with a warning for now. I'll also have a look at the overall consistency in the package in my upcoming PR.

Reply:
I've removed line 30 in a new commit. This line is not needed as the if-statement already checks if `raw_response` is `FALSE`. We should discuss the arguments further in a new issue. The user can input several arguments that will be internally inconsistent (and logically inconsistent for that matter), e.g. `get_municipality(2020:2022, simplify = FALSE, add_county = TRUE, raw_response = TRUE)`. I'm closing this discussion for now, but we can revisit it soon.